Automattic/vip-go-mu-plugins
# Bump Block Data API integration version to 1.1

## Description

We've released [version 1.1 of the Block Data API plugin](https://github.com/Automattic/vip-block-data-api/releases/tag/1.1.0) which includes GraphQL support.

The `1.1` version of the plugin has been successfully [pulled in by `vip-go-mu-plugins-ext`](https://github.com/Automattic/vip-go-mu-plugins-ext/tree/trunk/vip-integrations/vip-block-data-api-1.1), and we'd like to update the integration to point to the latest version.

## Changelog Description

### Integration Updated: VIP Block Data API

We upgraded the VIP Block Data API integration from `1.0.3` to `1.1`. This release adds GraphQL support.

## Pre-review checklist

Please make sure the items below have been covered before requesting a review:

- [ x This change uses a rollout method to ease with deployment (if applicable - especially for large scale actions that require writes).

## Pre-deploy checklist


## Steps to Test

Because there is a release freeze currently, testing this change requires some additional manual work.

1. Check out the latest version of [vip-go-mu-plugins-built](https://github.com/Automattic/vip-go-mu-plugins-built):

    ```bash
    git clone git@github.com:Automattic/vip-go-mu-plugins-built.git
    ```

2. In a separate directory, check out the latest version of [vip-go-mu-plugins-ext](https://github.com/Automattic/vip-go-mu-plugins-ext):

    ```bash
    git clone git@github.com:Automattic/vip-go-mu-plugins-ext.git
    ```

3. In the `vip-go-mu-plugins-ext` repository, copy over the [vip-integrations/vip-block-data-api-1.1
](https://github.com/Automattic/vip-go-mu-plugins-ext/tree/trunk/vip-integrations/vip-block-data-api-1.1) directory into `vip-go-mu-plugins-built`. The result should be that `vip-go-mu-plugins-built` contains a copied `vip-integrations/vip-block-data-api-1.1` folder.

4. Copy over the changes from this PR (a one-line change in `integrations/block-data-api.php` to `1.1`) into `vip-go-mu-plugins-built`.

5. Update a local `vip dev-env` instance to use the built mu-plugins folder:

    ```bash
    vip dev-env update --mu-plugins=~/projects/vip-go-mu-plugins-built
    ```

6. Enable the VIP Block Data API integration. In `client-mu-plugins/plugin-loader.php`, add:

    ```php
    \Automattic\VIP\Integrations\activate( 'block-data-api' );
    ```

7. This plugin update hooks into a filter in the [WPGraphQL plugin](https://wordpress.org/plugins/wp-graphql/). To test it, also install WPGraphQL locally and activate it.

8. Log in as an administrator. Go to Admin -> GraphQL -> GraphQL IDE. In the query composer, add this query:

    ```graphql
    query NewQuery {
      post(id: 1, idType: DATABASE_ID) {
        blocksData {
          blocks {
            attributes {
              name
              value
            }
            id
            name
            innerBlocks {
              attributes {
                name
                value
              }
              id
              name
              parentId
            }
          }
        }
      }
    }
    ```

    Replace the post ID `1` with a valid post ID if necessary.

9. Click the play button to run the query. GraphQL results containing block data should be visible:

    ![Screenshot 2023-12-14 at 11 37 20 AM](https://github.com/Automattic/vip-go-mu-plugins/assets/17870752/ca8a6f90-8966-46e0-8a88-faf29364dd62)
